### ISSUE TITLE
Firmware still too large, even after disabling features

### ISSUE THREAD

**Issue by ashleysommer** [vial-kb/vial-qmk#260]
Please forgive me if I'm doing something obvious wrong.

I'm compiling vial-enabled QMK for my Avalanche v4 keyboard, and I've been following the [guide here](https://get.vial.today/docs/firmware-size.html).

I am using a Pro Micro mcu with 28627 bytes of program memory available. My initial compilation was over 40k.
After turning off all optional Vial features, and enabling LTO, the resulting hex file is still too big:

> ` * The firmware is too large! 30352/28672 (1680 bytes over)`

My `keymaps/vial/rules.mk` file:
> ```
> VIA_ENABLE = yes
> VIAL_ENABLE = yes
> RGBLIGHT_ENABLE = yes
> ENCODER_MAP_ENABLE = yes
> LTO_ENABLE = yes
> QMK_SETTINGS = no
> TAP_DANCE_ENABLE = no
> COMBO_ENABLE = no
> KEY_OVERRIDE_ENABLE = no
> ```
My  `keymaps/vial/rules.mk` file:
> ```
> #pragma once
> #define VIAL_KEYBOARD_UID {0x23, 0x3D, 0x32, 0xDD, 0xC0, 0x95, 0x33, 0xF4}
> #define VIAL_UNLOCK_COMBO_ROWS { 0, 1 }
> #define VIAL_UNLOCK_COMBO_COLS { 0, 5 }
> #define VIAL_COMBO_ENTRIES 1
> #define VIAL_KEY_OVERRIDE_ENTRIES 1
> #define DYNAMIC_KEYMAP_LAYER_COUNT 1
> ```

Is it because I'm using ENCODER_MAP? Or due to RGBLIGHT_ENABLE?

**Comment by ashleysommer** [comment]
I've confirmed that LTO is working, because when I disable it, the firmware is larger:
> `The firmware is too large! 33520/28672 (4848 bytes over)`

**Comment by ashleysommer** [comment]
I did some more tweaking using the tweaks suggested in the [QMK squeezing space from AVR page](https://docs.qmk.fm/#/squeezing_avr), that got me closer, but still not enough.

I disabled RGBLIGHT, that free'd up plenty of room, and I was able to create a success image. But I do need RGBLight working, I disabled a bunch of the animation types (as suggested in the QMK doc) and in the end I also had remove support for the OLED screen on this build, in order to get the Vial features I want.
